_overviews/collections-2.13/conversion-between-option-and-collection.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: multipage-overview
+title: Conversion Between Option and Collection
+partof: collections-213
+overview-name: Collections
+
+num: 18
+previous-page: conversion-between-java-and-scala-collections
+
+permalink: /overviews/collections-2.13/:title.html
+---
+`Option` can be seen as a collection that has zero or exactly one element, and it provides a degree of interoperability with the collection types found in the package `scala.collection`. In particular, it implements the interface `IterableOnce`, which models the simplest form of collections: something that can be iterated over, at least once. However, `Option` does not implement the more comprehensive interface of `Iterable`. Indeed, we cannot provide a sensible implementation for the operation [`fromSpecific`](https://github.com/scala/scala/blob/6c68c2825e893bb71d6dc78465ac8c6f415cbd93/src/library/scala/collection/Iterable.scala#L173), which is supposed to create an `Option` from a collection of possibly more than one element. Starting from [Scala 2.13](https://github.com/scala/scala/pull/8038), `Option` was made an `IterableOnce` but not an `Iterable`.
+
+Hence `Option` can be used everywhere an `IterableOnce` is expected, for example, when calling `flatMap` on a collection (or inside a for-comprehension)
+
+```
+scala> for {
+     |   a <- Set(1)
+     |   b <- Option(41)
+     | } yield (a + b)
+val res0: scala.collection.immutable.Set[Int] = Set(42)
+``` 
```

Comment:
Can we use mdoc to typecheck and compute the result of this snippet? I couldn’t find how to do that in README.md maybe another contributor can help us?

Reply:
I was able to [process single markdown file](https://scalameta.org/mdoc/docs/installation.html#process-single-markdown-file) locally like so

```
coursier launch org.scalameta:mdoc_2.12:2.2.20 -- --in _overviews/collections-2.13/conversion-between-option-and-the-collections.md
```

This generates output markdown file with 

```scala
Some(42).drop(1)
// res1: Iterable[Int] = List()
```

Should I commit these outputs?

Reply:
No, the output of mdoc should be automatically displayed on the rendered website. Did you try to run the website locally? https://github.com/scala/docs.scala-lang#quickstart

Reply:
I tried running but mdoc evaluation does not show up. It seems currently even in production mdoc fences are not rendered evaluated, for example, see how [for-comprehensions.md](https://raw.githubusercontent.com/scala/docs.scala-lang/main/_tour/for-comprehensions.md) is rendered at https://docs.scala-lang.org/tour/for-comprehensions.html

Reply:
Oh, good point, maybe they are just typechecked, but not rendered… Maybe there is a modifier to use to display the result?

Reply:
It looks like they are only typechecked according to https://github.com/scala/docs.scala-lang/blob/main/contribute.md#code-blocks

Reply:
Thanks for investigating! Then I think what we can do is to show the output as an (unchecked) comment:

~~~ scala
for {
  x <- Set(41)
  y <- Option(1)
} yield x + y
// : Set[Int] = Set(42)
~~~

Reply:
Done. As a side note, seems because similar workarounds are used throughout docs to show outputs it would be hard to ever switch on automatic mdoc output rendering since it would effectively display duplicate outputs.